src/sumlog.jl

```diff
@@ -0,0 +1,36 @@
+using IrrationalConstants: logtwo
+
+"""
+$(SIGNATURES)
+
+Compute `sum(log.(X))`.
+
+`sum(log.(X))` can be evaluated much more quickly as `sum(log, X)`. However,
+this still requires computing `log` for each element of `X`. 
+
+`sumlog(X)` can be faster still, especially an the length of `X` increases.
+
+This works by representing the `j`th element of `X` as `xⱼ = aⱼ * 2 ^ bⱼ`,
+allowing us to write
+
+    ∑ⱼ log(xⱼ) = log(∏ⱼ aⱼ) + log(2) * ∑ⱼ bⱼ
+
```

Comment:
````suggestion
```math
\\sum_j \\log{x_j} = \\log(\\prod_j a_j) + \\log{2} \\sum_j b_j
```
````

Reply:
Personally, I tend to read docstrings from the REPL, where LaTeX doesn't render. But we can change it if you have a strong preference this way

Reply:
In this package we use math-blocks and double ticks with LaTeX syntax for equations, so we should be consistent here. AFAIK this is also recommended in the Julia documentation and works better with Documenter.